graf3d/eve7/inc/ROOT/REveBoxSet.hxx

```diff
@@ -0,0 +1,109 @@
+// @(#)root/eve:$Id$
+// Authors: Matevz Tadel & Alja Mrak-Tadel: 2006, 2007
+
+/*************************************************************************
+ * Copyright (C) 1995-2007, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#ifndef ROOT_REveBoxSet
+#define ROOT_REveBoxSet
+
+#include "ROOT/REveDigitSet.hxx"
+#include "ROOT/REveVector.hxx"
+
+class TGeoMatrix;
+class TRandom;
+
+namespace ROOT {
+namespace Experimental {
+class REveBoxSet: public REveDigitSet
```

Comment:
Would a templated class here make sense ? The box type is fixed for the entire box set right ?

Reply:
@rete REveDigitset uses REveChunkManager template to make optimize memory usage for large sets (e.g. hounded thousand digits). Maybe REveBoxSet is not the best class name because it is a set of generic 3D objects.